tests/sites/sidebars/_templates_single_sidebar/components/sidebar-nav-bs.html

```diff
@@ -1,6 +1,6 @@
 <nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
   <div class="bd-toc-item active">
     <!-- Specify a startdepth of 0 instead of default of 1 -->
-    {{ generate_nav_html("sidebar", startdepth=0, maxdepth=4, collapse=True, includehidden=True, titles_only=True) }}
+    {{ generate_toctree_html("sidebar", startdepth=0, maxdepth=4, collapse=True, includehidden=True, titles_only=True) }}
```

Comment:
FYI this will be a breaking change at least for SciPy since we are using this in a template (`sidebar-nav-bs.html`). Putting the template here in case we do something really wrong.

```
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  <div class="bd-toc-item active">
    {% if pagename.startswith("reference") %}
    {{ generate_nav_html("sidebar", maxdepth=1, collapse=True, includehidden=True, titles_only=True) }}
    {% else %}
    {{ generate_nav_html("sidebar", maxdepth=2, collapse=True, includehidden=True, titles_only=True) }}
    {% endif %}
  </div>
</nav>
```

Reply:
ah, we can change this back if need be, I just thought that `toctree_html` was a more descriptive name. Maybe we can just add a deprecation warning

Reply:
As I either way need to adapt more things I am personally fine with this change. But yes if this was supposed to be public, having a warning would be nice.

Reply:
By the way, can this template be the cause of this formatting issue I now see:
<img width="390" alt="Screenshot 2022-08-17 at 16 40 03" src="https://user-images.githubusercontent.com/23188539/185166979-d3655d0d-0853-42b0-9e08-27274550d19f.png">

On 0.9 it's fine but not on the RC or here.

Reply:
there's now a deprecation warning for this function, and I can't see this checkbox issue on the scipy docs, so I'll resolve this and can you open a new issue with a reproducible example of this issue on the scipy docs if it's still there? (w/ the extra checkbox)